apps/src/craft/code-connection/craft.js

```diff
@@ -218,7 +218,11 @@ export const executeUserCode = function (client, code) {
 
   // Register async methods
   codegen.asyncFunctionList = Object.values(asyncMethods);
-  interpreter = CustomMarshalingInterpreter.evalWith(code, Object.assign(asyncMethods, methods));
+  interpreter = CustomMarshalingInterpreter.evalWith(
+    code,
+    Object.assign(asyncMethods, methods),
+    {asyncFunctionList: asyncMethods}
```

Comment:
1. In `evalWith` you clearly mean to treat `asyncMethods` as an array, but it looks like it's an object here (per the `Object.values(asyncMethods)` call above).
2. Even if it was an array, the `asyncMethods` list here isn't the same set we were assigning to `codegen.asyncFucntionList` before because the `Object.assign` call on the previous line mutates `asyncMethods`.  This wasn't a problem before because we called `Object.values` before mutating `asyncMethods` but now I think you might be treating _all_ methods as async.

Is this a mistake?